s3/s3_transport/include/s3_transport.hpp

```diff
@@ -573,11 +582,35 @@ namespace irods::experimental::io::s3_transport
 
                 // use multipart if we have multiple client transfer threads or if the object size is > 2 * minimum part size
                 if ( use_streaming_multipart() ) {
-                    begin_part_upload_thread_ptr_ = std::make_unique<std::thread>(
-                            &s3_transport::s3_upload_part_worker_routine, this, false, 0, 0, get_file_offset());
+                    try {
+                        begin_part_upload_thread_ptr_ = std::make_unique<std::thread>(
+                                &s3_transport::s3_upload_part_worker_routine, this, false, 0, 0, get_file_offset());
+                    } catch (const std::bad_alloc& ba) {
+                        std::stringstream error_msg;
+                        error_msg << "Allocation error when creating upload part thread. [" << ba.what() << "]";
+                        this->set_error(ERROR(S3_PUT_ERROR, error_msg.str().c_str()));
+                        return 0;
+                    } catch (const std::system_error& se) {
+                        std::stringstream error_msg;
+                        error_msg << "System error when creating upload part thread. [" << se.what() << "]";
```

Comment:
i'm not recommending this be changed now, but fmtlib is likely easier to use and possibly provides better performance than `std::stringstream`.

we should start moving away from `std::stringstream`.